packages/@uppy/thumbnail-generator/src/index.js

```diff
@@ -315,15 +322,43 @@ module.exports = class ThumbnailGenerator extends Plugin {
     })
   }
 
+  waitUntilAllProcessed = (fileIDs) => {
+    fileIDs.forEach((fileID) => {
+      const file = this.uppy.getFile(fileID)
+      this.uppy.emit('preprocess-progress', file, {
+        mode: 'indeterminate',
+        message: this.i18n('generatingThumbnails')
+      })
+    })
+
+    return new Promise((resolve, reject) => {
+      if (this.queueProcessing) {
+        this.uppy.on('thumbnail:all-generated', () => {
```

Comment:
should this be `this.uppy.once` instead? Is it possible that this specific listener will always be triggered for every future thumbnail generations that will happen through out the lifetime of the Uppy instance?

Reply:
Ife is right, every time we upload a new batch of files, this event listener is getting attached and executed. Which is not a big deal, because we only `resolve()` here, but using `.once()` looks cleaner.

Reply:
Agreed! Thanks!